### ISSUE TITLE
Timezone-naive timestamps are assigned the timezone of the period_class

### ISSUE THREAD

**Issue by tonyfresko** [ample-tech/electric-units#2]
An issue came up when using timezone-naive sensor readings. I went through the following process:

- Used the sensor values to create a list of timezone-naive `WattSample`s
- These were then aggregated together into an `ElectricalEnergy` object
- Split the energy use into settlement periods with the method `.by_period(NemSettlementPeriod)`

If there's no timezone info in the input timestamps, then they're assumed to be in the same timezone as the period_class (`NemSettlementPeriod`). This could lead to some unpredictable behaviour.

 My 3 thoughts for helping this situation where the input timestamps don't have timezone info:

1. Show a warning, saying that they will be assumed to be timezone of the given period_class
2. Show a warning AND assume UTC
3. Raise an exception

My preference is for option 2, as it gives more predictable behaviour, but I'm interested to know what you think.

**Comment by tomharvey** [comment]
It's the [`SettlementPeriod.localize`](https://github.com/ample-tech/electric-units/blob/master/electric_units/base_settlement_period.py#L92) method doing the work here - either converting a TZ aware datetime, or assuming it's already local if the datetime is TZ naive.

Raising an exception seems overly strict, but continuing to do nothing isn't very clear either.

I like the suggestion to use [Warnings](https://docs.python.org/3/library/warnings.html) to flag this the first time it happens - and allow users to silence future messages.

But, I don't agree that UTC should be assumed for TZ naive datetimes. In most instances we would be working with a `None` timezone and the timezone of the `SettlementPeriod` class - neither of which would be UTC (unless in the UK national grid). I think it's simpler to assume that the date times share a timezone, but it's not explicit when naive.

**Comment by tonyfresko** [comment]
Yes, that makes sense - it's probably more likely that timezone-naive timestamps will be in a local time rather than UTC.
